command/generate.go

```diff
@@ -59,10 +59,24 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 			catcher.Add(errors.Wrapf(err, "Problem reading from file '%s'", fn))
 			continue
 		}
-		post = append(post, data)
+		json = append(json, data)
 	}
 	if catcher.HasErrors() {
 		return errors.WithStack(catcher.Resolve())
 	}
+	post := makeJsonOfAllFiles(json)
 	return errors.Wrap(comm.GenerateTasks(ctx, td, post), "Problem posting task data")
 }
+
+// makeJsonOfAllFiles creates a single JSON document that is an array of all JSON files. This allows
+// us to avoid posting multiple JSON files.
+func makeJsonOfAllFiles(json [][]byte) []byte {
+	post := []byte(`[`)
+	for _, j := range json {
+		post = append(post, j...)
+		post = append(post, byte(','))
+	}
+	post = post[:len(post)-1] // strip trailing comma
+	post = append(post, byte(']'))
+	return post
+}
```

Comment:
it wouldn't be absurd to marshal the input into map[string]interface{} (which I think you can round trip) and then assemble a slice of those and then marshal it to avoid needing to do (even minor) json mangling by hand.

Reply:
For whatever reason I couldn't get this to work with the json parser. I also tried json.RawMessage.